### DISCUSSION TITLE
AzureRM PowerShell module is now deprecated

### DISCUSSION THREAD

**Discussion by angaaruriakhil** [actions/virtual-environments#9833] [General questions] [answered]
As per [this article](https://learn.microsoft.com/en-us/powershell/azure/quickstart-migrate-azurerm-to-az-automatically?view=azps-11.6.0) - "The AzureRM PowerShell module has been officially deprecated as of February 29, 2024. Users are advised to migrate from AzureRM to the Az PowerShell module to ensure continued support and updates."

Our users of our Windows 2022 images (built from this repository this month) get the following warnings: 

`WARNING: Both Az and AzureRM modules were detected on this machine. Az and AzureRM modules cannot be imported in the same session or used in the same script or runbook. If you are running PowerShell in an environment you control you can use the 'Uninstall-AzureRm' cmdlet to remove all AzureRm modules from your machine. If you are running in Azure Automation, take care that none of your runbooks import both Az and AzureRM modules. More information can be found here: https://aka.ms/azps-migration-guide
##[warning]You're using AzureRM which will be retired soon, please schedule an update.`

Are there any plans to gradually phase out/remove the AzureRM PowerShell module? And if so, when this may happen? We want our teams to be prepared to update any code ahead of time. 

Thanks in advance!

**Comment by erik-bershel** [comment, marked as answer]
Hey @angaaruriakhil!

We have no specific plans to remove this in the next couple of releases at least, but it will most likely be removed around August of this year due to general repository rules. But the situation may change in both directions if some critical vulnerability is identified or if some critical tools for a large number of users depend on this package. As in the case of the OpenSSL 1.1 library.

**Reply by angaaruriakhil** [reply]
Thanks kindly Erik for your confirmation.
